compaction/compaction_manager.cc

```diff
@@ -1602,6 +1602,51 @@ bool compaction_manager::requires_cleanup(table_state& t, const sstables::shared
 }
 
 future<> compaction_manager::perform_cleanup(owned_ranges_ptr sorted_owned_ranges, table_state& t) {
+    constexpr auto sleep_duration = std::chrono::seconds(1);
+    constexpr auto max_idle_duration = std::chrono::seconds(300);
+
+    size_t eligible = 1;    // try_perform_cleanup first time around
+    std::optional<seastar::lowres_clock::time_point> last_idle;
+
+    for (;;) {
```

Comment:
This could be `while (cs.sstables_requiring_cleanup.empty() && idle < max_idle_duration) {`.

It is so annoying when one has to hunt around for the exit conditions in a loop.

Reply:
The problem with that is that sstables_requiring_cleanup is empty on entry.
We'd need to `co_await try_perform_cleanup` before entering the loop.
Maybe it's not a bad idea.

Reply:
Agree